src/ivoc/ocjump.h

```diff
@@ -1,34 +1,43 @@
 #pragma once
 union Inst;
-class OcJumpImpl;
-struct Symlist;
 struct Object;
 union Objectdata;
+struct Symlist;
 
-class ObjectContext {
-  public:
+/** @brief How many NEURON try { ... } catch(...) { ... } blocks are in the call stack.
+ *
+ *  Errors inside NEURON are triggered using hoc_execerror, which ultimately
+ *  throws an exception. To replicate the old logic, we sometimes need to insert
+ *  a try/catch block *only if* there is no try/catch block less deeply nested
+ *  on the call stack. This global variable tracks how many such blocks are
+ *  currently present on the stack.
+ */
+extern int nrn_try_catch_nest_depth;
```

Comment:
This is a reasonably simple approach that tries to preserve the old logic, but it is not a 100% match.